paymentapi/src/main/java/fi/hel/verkkokauppa/payment/service/PaymentPaytrailService.java

```diff
@@ -0,0 +1,72 @@
+package fi.hel.verkkokauppa.payment.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import fi.hel.verkkokauppa.common.configuration.ServiceConfigurationKeys;
+import fi.hel.verkkokauppa.common.rest.CommonServiceConfigurationClient;
+import fi.hel.verkkokauppa.payment.api.data.PaymentMethodDto;
+import fi.hel.verkkokauppa.payment.constant.GatewayEnum;
+import fi.hel.verkkokauppa.payment.paytrail.factory.PaytrailAuthClientFactory;
+import lombok.extern.slf4j.Slf4j;
+import org.helsinki.paytrail.PaytrailClient;
+import org.helsinki.paytrail.mapper.PaytrailPaymentMethodsResponseMapper;
+import org.helsinki.paytrail.request.paymentmethods.PaytrailPaymentMethodsRequest;
+import org.helsinki.paytrail.response.paymentmethods.PaytrailPaymentMethodsResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+@Service
+@Slf4j
+public class PaymentPaytrailService {
+
+    private final PaytrailAuthClientFactory paytrailClientFactory;
+    private final CommonServiceConfigurationClient commonServiceConfigurationClient;
+    private final PaytrailPaymentMethodsResponseMapper paymentMethodsResponseMapper;
+
+    @Autowired
+    PaymentPaytrailService(
+            PaytrailAuthClientFactory paytrailClientFactory,
+            CommonServiceConfigurationClient commonServiceConfigurationClient,
+            ObjectMapper mapper
+    ) {
+        this.paytrailClientFactory = paytrailClientFactory;
+        this.commonServiceConfigurationClient = commonServiceConfigurationClient;
+        this.paymentMethodsResponseMapper = new PaytrailPaymentMethodsResponseMapper(mapper);
+    }
+
+    public PaymentMethodDto[] getOnlinePaymentMethodList(String merchantId, String namespace, String currency) {
+        if (merchantId != null && !merchantId.isEmpty()) {
+            String shopId = commonServiceConfigurationClient.getMerchantConfigurationValue(merchantId, namespace, ServiceConfigurationKeys.MERCHANT_SHOP_ID);
+            if (shopId != null && !shopId.isEmpty()) {
```

Comment:
StringUtils.isNotEmpty could be used in here

Reply:
Aaa I did not notice there is a util for this, I'll use it.